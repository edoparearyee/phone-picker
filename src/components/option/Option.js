import React, { Component } from 'react';
import './Option.css';

class Option extends Component {
  render() {
    const activeClass = this.props.active
      ? 'Option--active'
      : 'Option--inactive';

    const colorClass = this.props.color ? 'Option--color' : '';
    return (
      <li className={`Option ${activeClass} ${colorClass}`}>
        <button
          className="Option__action"
          style={{ backgroundColor: this.props.color }}
          onClick={() => this.props.action(this.props.value)}
        >
          {this.props.children}
        </button>
      </li>
    );
  }
}

export default Option;
