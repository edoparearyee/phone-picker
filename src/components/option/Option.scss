@import '../../styles/variables';
@import '../../styles/functions';

.Option {
  border: rem(2px) solid transparent;
  border-radius: 25%;
  display: inline-flex;
  list-style: none;
  padding: rem(5px);
}

.Option--active {
  border-color: $option-outer-border-color;
}

.Option__action {
  -webkit-appearance: none;
  appearance: none;
  align-items: center;
  background-color: $option-inner-bg-color;
  border: rem(1px) solid $option-inner-border-color;
  border-radius: 20%;
  cursor: pointer;
  display: inline-flex;
  justify-content: center;
  outline: none;
  overflow: hidden;
  height: rem(46px);
  position: relative;
  width: rem(46px);

  .Option--color &::after {
    background-image: linear-gradient(
      to bottom,
      transparent,
      rgba($black, 0.3)
    );
    bottom: 0;
    content: '';
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    z-index: 1;
  }
}
