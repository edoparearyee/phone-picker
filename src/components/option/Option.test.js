import React from 'react';
import ReactDOM from 'react-dom';
import Option from './Option';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Option />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Option has class Option--inactive', () => {
  const component = renderer.create(<Option active={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Option has class Option--active', () => {
  const component = renderer.create(<Option active={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Option has class Option--inactive and content', () => {
  const component = renderer.create(<Option active={false}>64</Option>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Option has class Option--active and content', () => {
  const component = renderer.create(<Option active={true}>64</Option>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
