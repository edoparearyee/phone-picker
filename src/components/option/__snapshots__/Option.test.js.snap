// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option has class Option--active 1`] = `
<li
  className="Option Option--active "
>
  <button
    className="Option__action"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />
</li>
`;

exports[`Option has class Option--active and content 1`] = `
<li
  className="Option Option--active "
>
  <button
    className="Option__action"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    64
  </button>
</li>
`;

exports[`Option has class Option--active and custom color 1`] = `
<li
  className="Option Option--active Option--color"
>
  <button
    className="Option__action"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f8e7dc",
      }
    }
  />
</li>
`;

exports[`Option has class Option--inactive 1`] = `
<li
  className="Option Option--inactive "
>
  <button
    className="Option__action"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />
</li>
`;

exports[`Option has class Option--inactive and content 1`] = `
<li
  className="Option Option--inactive "
>
  <button
    className="Option__action"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    64
  </button>
</li>
`;

exports[`Option has custom color 1`] = `
<li
  className="Option Option--inactive Option--color"
>
  <button
    className="Option__action"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f8e7dc",
      }
    }
  />
</li>
`;
